tests/tokenizers_test.py

```diff
@@ -468,15 +468,30 @@ def check_tokenizer_output(
     skip_missing_outputs: bool = False,
     hf_tokenizer_kwargs: Optional[Dict[str, Any]] = None,
     calculate_diff: bool = False,
+    add_special_tokens_state_flag: Optional[bool] = None,
 ) -> None:
     hf_tokenizer, ov_tokenizer = tokenizers
     hf_tokenizer_kwargs = {} if hf_tokenizer_kwargs is None else hf_tokenizer_kwargs
 
     if isinstance(test_string, str):
         test_string = [test_string]
-
+    
     hf_tokenized = hf_tokenizer(test_string, return_tensors="np", truncation=True, **hf_tokenizer_kwargs)
-    ov_tokenized = ov_tokenizer(test_string)
+    # If add_special_tokens_state_flag is defined, set it's value to state
+    if add_special_tokens_state_flag is not None:
+        hf_tokenizer_kwargs['add_special_tokens'] = add_special_tokens_state_flag
+        ov_infer_request = ov_tokenizer.create_infer_request()
+        states = ov_infer_request.query_state()
+        state_tensor = ov.Tensor(np.array([add_special_tokens_state_flag], dtype=np.int32), ov.Shape([]))
+        for state in states:
+            if state.name != SPECIAL_TOKENS_STATE_NAME:
+                continue
+            state.state = state_tensor
+        
+        ov_infer_request.set_input_tensor(ov.Tensor(test_string))
+        ov_tokenized = ov_infer_request.infer()
```

Comment:
You can query CompiledModel for a state and change it, and then just calling it as usual will be enough.
```python
if add_special_tokens_state_flag is not None:
    states = ov_tokenizer.query_state()
    for state in states:
        if state.name == SPECIAL_TOKENS_STATE_NAME:
            state.state = state_tensor
ov_tokenized = ov_tokenizer(test_string)
```

Reply:
done